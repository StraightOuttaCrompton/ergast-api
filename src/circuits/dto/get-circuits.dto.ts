import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";

export class GetCircuitsDto {
    /**
     * TODO: extract into class which this inherits from
     */
    @IsOptional()
    @IsInt()
    @IsPositive()
    offset: number;

    @IsOptional()
    @IsInt()
    @IsPositive()
    limit: number;
    /**
     * --------------------------------------------------
     */

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
