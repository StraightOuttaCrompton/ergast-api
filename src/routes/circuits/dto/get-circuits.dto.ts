import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetCircuitsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
