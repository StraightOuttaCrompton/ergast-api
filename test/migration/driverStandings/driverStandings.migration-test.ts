import querystring from "querystring";
import { GetDriverStandingsDto } from "src/routes/driverStandings/dto/get-driver-standings.dto";
import { getMigrationTest } from "../migrationUtils";
import path from "path";

function convertLegacyDriverStandingsResponse(driverStanding: any) {
    const { Constructors, ...rest } = driverStanding;

    return rest;
}

function convertLegacyResponse(response: any) {
    const { season, round, DriverStandings } = response;

    return { season, round, driverStandings: DriverStandings.map(convertLegacyDriverStandingsResponse) };
}

describe("GET /driverStandings smoke tests", () => {
    const migrationTest = getMigrationTest(__dirname, path.basename(__filename), (response) => {
        return response.MRData.StandingsTable.StandingsLists.map(convertLegacyResponse);
    });

    const endpoint = "/driverStandings";

    const queries: { queryParams: Partial<GetDriverStandingsDto>; legacyRoute: string }[] = [
        {
            queryParams: { year: 2022 },
            legacyRoute: "2022",
        },
        {
            queryParams: { year: 2009, round: 5 },
            legacyRoute: "2009/5",
        },
        {
            queryParams: { year: 2012, round: 3, driverId: "vettel" },
            legacyRoute: "2012/3/drivers/vettel",
        },
        {
            queryParams: { driverId: "hamilton" },
            legacyRoute: "drivers/hamilton",
        },
        {
            queryParams: { year: 2019, driverId: "hamilton" },
            legacyRoute: "2019/drivers/hamilton",
        },
    ];

    migrationTest(endpoint, `https://ergast.com/api/f1/driverStandings.json`);

    migrationTest(
        `${endpoint}?${querystring.stringify({ position: 2, year: 2022 })}`,
        `https://ergast.com/api/f1/2022/driverStandings/2.json`
    );

    queries.forEach(({ queryParams, legacyRoute }) => {
        migrationTest(
            `${endpoint}?${querystring.stringify(queryParams)}`,
            `https://ergast.com/api/f1/${legacyRoute}/driverStandings.json`
        );
    });
});
