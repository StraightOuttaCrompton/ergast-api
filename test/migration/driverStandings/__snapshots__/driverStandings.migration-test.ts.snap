// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /driverStandings smoke tests /driverStandings 1`] = `
[
  {
    "driverStandings": [
      {
        "Driver": {
          "dateOfBirth": "1906-10-30",
          "driverId": "farina",
          "familyName": "Farina",
          "givenName": "Nino",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Nino_Farina",
        },
        "points": "30",
        "position": "1",
        "positionText": "1",
        "wins": "3",
      },
      {
        "Driver": {
          "dateOfBirth": "1911-06-24",
          "driverId": "fangio",
          "familyName": "Fangio",
          "givenName": "Juan",
          "nationality": "Argentine",
          "url": "http://en.wikipedia.org/wiki/Juan_Manuel_Fangio",
        },
        "points": "27",
        "position": "2",
        "positionText": "2",
        "wins": "3",
      },
      {
        "Driver": {
          "dateOfBirth": "1898-06-09",
          "driverId": "fagioli",
          "familyName": "Fagioli",
          "givenName": "Luigi",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Luigi_Fagioli",
        },
        "points": "24",
        "position": "3",
        "positionText": "3",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1905-11-05",
          "driverId": "rosier",
          "familyName": "Rosier",
          "givenName": "Louis",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Louis_Rosier",
        },
        "points": "13",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1918-07-13",
          "driverId": "ascari",
          "familyName": "Ascari",
          "givenName": "Alberto",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Alberto_Ascari",
        },
        "points": "11",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1918-07-04",
          "driverId": "parsons",
          "familyName": "Parsons",
          "givenName": "Johnnie",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Johnnie_Parsons",
        },
        "points": "9",
        "position": "6",
        "positionText": "6",
        "wins": "1",
      },
      {
        "Driver": {
          "dateOfBirth": "1907-12-18",
          "driverId": "holland",
          "familyName": "Holland",
          "givenName": "Bill",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Bill_Holland",
        },
        "points": "6",
        "position": "7",
        "positionText": "7",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1914-07-15",
          "driverId": "bira",
          "familyName": "Bira",
          "givenName": "Prince",
          "nationality": "Thai",
          "url": "http://en.wikipedia.org/wiki/Prince_Bira",
        },
        "points": "5",
        "position": "8",
        "positionText": "8",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1914-11-12",
          "driverId": "whitehead",
          "familyName": "Whitehead",
          "givenName": "Peter",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Peter_Whitehead_(racing_driver)",
        },
        "points": "4",
        "position": "9",
        "positionText": "9",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1899-08-03",
          "driverId": "chiron",
          "familyName": "Chiron",
          "givenName": "Louis",
          "nationality": "Monegasque",
          "url": "http://en.wikipedia.org/wiki/Louis_Chiron",
        },
        "points": "4",
        "position": "10",
        "positionText": "10",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1911-07-02",
          "driverId": "reg_parnell",
          "familyName": "Parnell",
          "givenName": "Reg",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Reg_Parnell",
        },
        "points": "4",
        "position": "11",
        "positionText": "11",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1906-05-26",
          "driverId": "rose",
          "familyName": "Rose",
          "givenName": "Mauri",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Mauri_Rose",
        },
        "points": "4",
        "position": "12",
        "positionText": "12",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1909-07-22",
          "driverId": "serafini",
          "familyName": "Serafini",
          "givenName": "Dorino",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Dorino_Serafini",
        },
        "points": "3",
        "position": "13",
        "positionText": "13",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1904-10-08",
          "driverId": "cabantous",
          "familyName": "Cabantous",
          "givenName": "Yves",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Yves_Giraud_Cabantous",
        },
        "points": "3",
        "position": "14",
        "positionText": "14",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1906-08-31",
          "driverId": "sommer",
          "familyName": "Sommer",
          "givenName": "Raymond",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Raymond_Sommer",
        },
        "points": "3",
        "position": "15",
        "positionText": "15",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1917-04-12",
          "driverId": "manzon",
          "familyName": "Manzon",
          "givenName": "Robert",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Robert_Manzon",
        },
        "points": "3",
        "position": "16",
        "positionText": "16",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1919-09-30",
          "driverId": "green",
          "familyName": "Green",
          "givenName": "Cecil",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Cecil_Green",
        },
        "points": "3",
        "position": "17",
        "positionText": "17",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1896-12-28",
          "driverId": "etancelin",
          "familyName": "Étancelin",
          "givenName": "Philippe",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Philippe_%C3%89tancelin",
        },
        "points": "3",
        "position": "18",
        "positionText": "18",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1903-06-09",
          "driverId": "bonetto",
          "familyName": "Bonetto",
          "givenName": "Felice",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Felice_Bonetto",
        },
        "points": "2",
        "position": "19",
        "positionText": "19",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1907-04-12",
          "driverId": "chaboud",
          "familyName": "Chaboud",
          "givenName": "Eugène",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Eug%C3%A8ne_Chaboud",
        },
        "points": "1",
        "position": "20",
        "positionText": "20",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1916-09-12",
          "driverId": "bettenhausen",
          "familyName": "Bettenhausen",
          "givenName": "Tony",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Tony_Bettenhausen",
        },
        "points": "1",
        "position": "21",
        "positionText": "21",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1912-04-14",
          "driverId": "chitwood",
          "familyName": "Chitwood",
          "givenName": "Joie",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Joie_Chitwood",
        },
        "points": "1",
        "position": "22",
        "positionText": "22",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1914-05-18",
          "driverId": "graffenried",
          "familyName": "de Graffenried",
          "givenName": "Toulo",
          "nationality": "Swiss",
          "url": "http://en.wikipedia.org/wiki/Toulo_de_Graffenried",
        },
        "points": "0",
        "position": "23",
        "positionText": "23",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1914-01-19",
          "driverId": "gerard",
          "familyName": "Gerard",
          "givenName": "Bob",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Bob_Gerard",
        },
        "points": "0",
        "position": "24",
        "positionText": "24",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1909-05-16",
          "driverId": "villoresi",
          "familyName": "Villoresi",
          "givenName": "Luigi",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Luigi_Villoresi",
        },
        "points": "0",
        "position": "25",
        "positionText": "25",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1909-08-27",
          "driverId": "pozzi",
          "familyName": "Pozzi",
          "givenName": "Charles",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Charles_Pozzi",
        },
        "points": "0",
        "position": "26",
        "positionText": "26",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1910-09-07",
          "driverId": "wallard",
          "familyName": "Wallard",
          "givenName": "Lee",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Lee_Wallard",
        },
        "points": "0",
        "position": "27",
        "positionText": "27",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1916-08-11",
          "driverId": "claes",
          "familyName": "Claes",
          "givenName": "Johnny",
          "nationality": "Belgian",
          "url": "http://en.wikipedia.org/wiki/Johnny_Claes",
        },
        "points": "0",
        "position": "28",
        "positionText": "28",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1905-12-22",
          "driverId": "levegh",
          "familyName": "Levegh",
          "givenName": "Pierre",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Pierre_Levegh",
        },
        "points": "0",
        "position": "29",
        "positionText": "29",
        "wins": "0",
      },
      {
        "Driver": {
          "dateOfBirth": "1906-07-06",
          "driverId": "harrison",
          "familyName": "Harrison",
          "givenName": "Cuth",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cuth_Harrison",
        },
        "points": "0",
        "position": "30",
        "positionText": "30",
        "wins": "0",
      },
    ],
    "round": "7",
    "season": "1950",
  },
]
`;

exports[`GET /driverStandings smoke tests /driverStandings?driverId=hamilton 1`] = `
[
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "109",
        "position": "2",
        "positionText": "2",
        "wins": "4",
      },
    ],
    "round": "17",
    "season": "2007",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "98",
        "position": "1",
        "positionText": "1",
        "wins": "5",
      },
    ],
    "round": "18",
    "season": "2008",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "49",
        "position": "5",
        "positionText": "5",
        "wins": "2",
      },
    ],
    "round": "17",
    "season": "2009",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "240",
        "position": "4",
        "positionText": "4",
        "wins": "3",
      },
    ],
    "round": "19",
    "season": "2010",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "227",
        "position": "5",
        "positionText": "5",
        "wins": "3",
      },
    ],
    "round": "19",
    "season": "2011",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "190",
        "position": "4",
        "positionText": "4",
        "wins": "4",
      },
    ],
    "round": "20",
    "season": "2012",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "189",
        "position": "4",
        "positionText": "4",
        "wins": "1",
      },
    ],
    "round": "19",
    "season": "2013",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "384",
        "position": "1",
        "positionText": "1",
        "wins": "11",
      },
    ],
    "round": "19",
    "season": "2014",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "381",
        "position": "1",
        "positionText": "1",
        "wins": "10",
      },
    ],
    "round": "19",
    "season": "2015",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "380",
        "position": "2",
        "positionText": "2",
        "wins": "10",
      },
    ],
    "round": "21",
    "season": "2016",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "363",
        "position": "1",
        "positionText": "1",
        "wins": "9",
      },
    ],
    "round": "20",
    "season": "2017",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "408",
        "position": "1",
        "positionText": "1",
        "wins": "11",
      },
    ],
    "round": "21",
    "season": "2018",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "413",
        "position": "1",
        "positionText": "1",
        "wins": "11",
      },
    ],
    "round": "21",
    "season": "2019",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "347",
        "position": "1",
        "positionText": "1",
        "wins": "11",
      },
    ],
    "round": "17",
    "season": "2020",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "387.5",
        "position": "2",
        "positionText": "2",
        "wins": "8",
      },
    ],
    "round": "22",
    "season": "2021",
  },
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "240",
        "position": "6",
        "positionText": "6",
        "wins": "0",
      },
    ],
    "round": "22",
    "season": "2022",
  },
]
`;

exports[`GET /driverStandings smoke tests /driverStandings?position=2&year=2022 1`] = `
[
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "LEC",
          "dateOfBirth": "1997-10-16",
          "driverId": "leclerc",
          "familyName": "Leclerc",
          "givenName": "Charles",
          "nationality": "Monegasque",
          "permanentNumber": "16",
          "url": "http://en.wikipedia.org/wiki/Charles_Leclerc",
        },
        "points": "308",
        "position": "2",
        "positionText": "2",
        "wins": "3",
      },
    ],
    "round": "22",
    "season": "2022",
  },
]
`;

exports[`GET /driverStandings smoke tests /driverStandings?year=2009&round=5 1`] = `
[
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "BUT",
          "dateOfBirth": "1980-01-19",
          "driverId": "button",
          "familyName": "Button",
          "givenName": "Jenson",
          "nationality": "British",
          "permanentNumber": "22",
          "url": "http://en.wikipedia.org/wiki/Jenson_Button",
        },
        "points": "41",
        "position": "1",
        "positionText": "1",
        "wins": "4",
      },
      {
        "Driver": {
          "code": "BAR",
          "dateOfBirth": "1972-05-23",
          "driverId": "barrichello",
          "familyName": "Barrichello",
          "givenName": "Rubens",
          "nationality": "Brazilian",
          "url": "http://en.wikipedia.org/wiki/Rubens_Barrichello",
        },
        "points": "27",
        "position": "2",
        "positionText": "2",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "VET",
          "dateOfBirth": "1987-07-03",
          "driverId": "vettel",
          "familyName": "Vettel",
          "givenName": "Sebastian",
          "nationality": "German",
          "permanentNumber": "5",
          "url": "http://en.wikipedia.org/wiki/Sebastian_Vettel",
        },
        "points": "23",
        "position": "3",
        "positionText": "3",
        "wins": "1",
      },
      {
        "Driver": {
          "code": "WEB",
          "dateOfBirth": "1976-08-27",
          "driverId": "webber",
          "familyName": "Webber",
          "givenName": "Mark",
          "nationality": "Australian",
          "url": "http://en.wikipedia.org/wiki/Mark_Webber_(racing_driver)",
        },
        "points": "15.5",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "TRU",
          "dateOfBirth": "1974-07-13",
          "driverId": "trulli",
          "familyName": "Trulli",
          "givenName": "Jarno",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Jarno_Trulli",
        },
        "points": "14.5",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "GLO",
          "dateOfBirth": "1982-03-18",
          "driverId": "glock",
          "familyName": "Glock",
          "givenName": "Timo",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Timo_Glock",
        },
        "points": "12",
        "position": "6",
        "positionText": "6",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "9",
        "position": "7",
        "positionText": "7",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "ALO",
          "dateOfBirth": "1981-07-29",
          "driverId": "alonso",
          "familyName": "Alonso",
          "givenName": "Fernando",
          "nationality": "Spanish",
          "permanentNumber": "14",
          "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
        },
        "points": "9",
        "position": "8",
        "positionText": "8",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "HEI",
          "dateOfBirth": "1977-05-10",
          "driverId": "heidfeld",
          "familyName": "Heidfeld",
          "givenName": "Nick",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Nick_Heidfeld",
        },
        "points": "6",
        "position": "9",
        "positionText": "9",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "ROS",
          "dateOfBirth": "1985-06-27",
          "driverId": "rosberg",
          "familyName": "Rosberg",
          "givenName": "Nico",
          "nationality": "German",
          "permanentNumber": "6",
          "url": "http://en.wikipedia.org/wiki/Nico_Rosberg",
        },
        "points": "4.5",
        "position": "10",
        "positionText": "10",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "KOV",
          "dateOfBirth": "1981-10-19",
          "driverId": "kovalainen",
          "familyName": "Kovalainen",
          "givenName": "Heikki",
          "nationality": "Finnish",
          "url": "http://en.wikipedia.org/wiki/Heikki_Kovalainen",
        },
        "points": "4",
        "position": "11",
        "positionText": "11",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "MAS",
          "dateOfBirth": "1981-04-25",
          "driverId": "massa",
          "familyName": "Massa",
          "givenName": "Felipe",
          "nationality": "Brazilian",
          "permanentNumber": "19",
          "url": "http://en.wikipedia.org/wiki/Felipe_Massa",
        },
        "points": "3",
        "position": "12",
        "positionText": "12",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "RAI",
          "dateOfBirth": "1979-10-17",
          "driverId": "raikkonen",
          "familyName": "Räikkönen",
          "givenName": "Kimi",
          "nationality": "Finnish",
          "permanentNumber": "7",
          "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
        },
        "points": "3",
        "position": "13",
        "positionText": "13",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "BUE",
          "dateOfBirth": "1988-10-31",
          "driverId": "buemi",
          "familyName": "Buemi",
          "givenName": "Sébastien",
          "nationality": "Swiss",
          "url": "http://en.wikipedia.org/wiki/S%C3%A9bastien_Buemi",
        },
        "points": "3",
        "position": "14",
        "positionText": "14",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "BOU",
          "dateOfBirth": "1979-02-28",
          "driverId": "bourdais",
          "familyName": "Bourdais",
          "givenName": "Sébastien",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/S%C3%A9bastien_Bourdais",
        },
        "points": "1",
        "position": "15",
        "positionText": "15",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "SUT",
          "dateOfBirth": "1983-01-11",
          "driverId": "sutil",
          "familyName": "Sutil",
          "givenName": "Adrian",
          "nationality": "German",
          "permanentNumber": "99",
          "url": "http://en.wikipedia.org/wiki/Adrian_Sutil",
        },
        "points": "0",
        "position": "16",
        "positionText": "16",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "PIQ",
          "dateOfBirth": "1985-07-25",
          "driverId": "piquet_jr",
          "familyName": "Piquet Jr.",
          "givenName": "Nelson",
          "nationality": "Brazilian",
          "url": "http://en.wikipedia.org/wiki/Nelson_Piquet,_Jr.",
        },
        "points": "0",
        "position": "17",
        "positionText": "17",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "KUB",
          "dateOfBirth": "1984-12-07",
          "driverId": "kubica",
          "familyName": "Kubica",
          "givenName": "Robert",
          "nationality": "Polish",
          "permanentNumber": "88",
          "url": "http://en.wikipedia.org/wiki/Robert_Kubica",
        },
        "points": "0",
        "position": "18",
        "positionText": "18",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "FIS",
          "dateOfBirth": "1973-01-14",
          "driverId": "fisichella",
          "familyName": "Fisichella",
          "givenName": "Giancarlo",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Giancarlo_Fisichella",
        },
        "points": "0",
        "position": "19",
        "positionText": "19",
        "wins": "0",
      },
      {
        "Driver": {
          "code": "NAK",
          "dateOfBirth": "1985-01-11",
          "driverId": "nakajima",
          "familyName": "Nakajima",
          "givenName": "Kazuki",
          "nationality": "Japanese",
          "url": "http://en.wikipedia.org/wiki/Kazuki_Nakajima",
        },
        "points": "0",
        "position": "20",
        "positionText": "20",
        "wins": "0",
      },
    ],
    "round": "5",
    "season": "2009",
  },
]
`;

exports[`GET /driverStandings smoke tests /driverStandings?year=2012&round=3&driverId=vettel 1`] = `
[
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "VET",
          "dateOfBirth": "1987-07-03",
          "driverId": "vettel",
          "familyName": "Vettel",
          "givenName": "Sebastian",
          "nationality": "German",
          "permanentNumber": "5",
          "url": "http://en.wikipedia.org/wiki/Sebastian_Vettel",
        },
        "points": "28",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
    ],
    "round": "3",
    "season": "2012",
  },
]
`;

exports[`GET /driverStandings smoke tests /driverStandings?year=2019&driverId=hamilton 1`] = `
[
  {
    "driverStandings": [
      {
        "Driver": {
          "code": "HAM",
          "dateOfBirth": "1985-01-07",
          "driverId": "hamilton",
          "familyName": "Hamilton",
          "givenName": "Lewis",
          "nationality": "British",
          "permanentNumber": "44",
          "url": "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        },
        "points": "413",
        "position": "1",
        "positionText": "1",
        "wins": "11",
      },
    ],
    "round": "21",
    "season": "2019",
  },
]
`;
